apptest/sharding_test.go

```diff
@@ -0,0 +1,82 @@
+package apptest
+
+import (
+	"fmt"
+	"math/rand/v2"
+	"testing"
+	"time"
+)
+
+func TestVminsertShardsDataVmselectBuildsFullResultFromShards(t *testing.T) {
+	defer testRemoveAll(t)
+
+	// Set up the following cluster configuration:
+	//
+	// - two vmstorage instances
+	// - vminsert points to the two vmstorages, its replication setting
+	//   is off which means it will only shard the incoming data across the two
+	//   vmstorages.
+	// - vmselect points to the two vmstorages and is expected to query both
+	//   vmstorages and build the full result out of the two partial results.
+
+	cli := newClient()
+	defer cli.closeConnections()
```

Comment:
Need to manually cleanup folder and connections seems fragile.
Can we introduce an abstraction that would take care of all this? E.g.:
```
	ts := newTestSuite(t)
	defer ts.close()

	cli := ts.cli
	vmstorage1 := mustStartVmstorage(t, "vmstorage-1", []string{
		"-storageDataPath=" + ts.Dir() + "/vmstorage-1",
	}, cli)
	defer vmstorage1.stop()
```